src/inmanta/ast/attribute.py

```diff
@@ -101,38 +107,35 @@ def validate(self, value: object) -> None:
         """
         if isinstance(value, Unknown):
             return
-        validation_type: Type = self.type
-        if self.is_multi():
-            validation_type = TypedList(validation_type)
-        if self.is_optional():
-            validation_type = NullableType(validation_type)
-        validation_type.validate(value)
+        self.type.validate(value)
 
     def get_new_result_variable(self, instance: "Instance", queue: QueueScheduler) -> ResultVariable:
-        if self.__multi:
-            mytype = TypedList(self.__type)
-        else:
-            mytype = self.__type
-
         out: ResultVariable["Instance"]
 
-        if self.__nullable:
+        if self.is_optional():
             # be a 0-1 relation
             self.end = None
             self.low = 0
             self.high = 1
             out = DeprecatedOptionVariable(self, instance, queue)
-            mytype = NullableType(mytype)
         else:
             out = ResultVariable()
 
-        out.set_type(mytype)
+        out.set_type(self.type)
         return out
 
     def is_optional(self) -> bool:
+        """
+            Returns true iff this attribute accepts null values.
+            Deprecated but still used internally.
+        """
         return self.__nullable
 
     def is_multi(self) -> bool:
+        """
+            Returns true iff this attribute expects a list of values of its type.
```

Comment:
This is not strictly correct with the updated type. 
```suggestion
            Returns true iff attributes types accepts a list of values of its basetype.
```

Reply:
Could you elaborate? I very intentionally used "expects" since it doesn't accept a single value of type `basetype`, right?

Reply:
I agree on your second suggestion by the way: `type` -> `basetype` is definitely more correct.

Reply:
ok, then take the good part and forget the rest ;-)